s2fft/utils/healpix_ffts.py

```diff
@@ -5,10 +5,24 @@
 import torch
 from s2fft.sampling import s2_samples as samples
 from functools import partial
+from s2fft.utils.jax_pritimive import BasePrimitive, register_primitive
+import jaxlib.mlir.ir as ir
+from s2fft_lib import _s2fft
+from jaxlib.hlo_helpers import custom_call
+from jax._src.lib.mlir.dialects import hlo
+from jax._src.interpreters import mlir
+from jax.core import Primitive, ShapedArray
+from jax.interpreters import ad, xla
+from jax.sharding import Mesh, NamedSharding
+from jax.sharding import PartitionSpec as P
+from typing import Tuple
+from jax._src.api import ShapeDtypeStruct
+from math import prod
```

Comment:
This appears to be unused.

Reply:
Multi-gpu + type hinting stuff.
Again, I can remove if necessary
Except for prod ... that was just a copy paste from my own work in jaxDecomp (will be removed)